File: lib/views/review_shift_sheet.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart' hide TextDirection;
import '../models/shift.dart';
import '../models/expense.dart';

class ReviewShiftSheet extends StatefulWidget {
  final Shift shift;
  final List<String> availableRoles;
  final Function(Shift) onSave;
  final VoidCallback onDiscard;

  const ReviewShiftSheet({
    super.key,
    required this.shift,
    required this.availableRoles,
    required this.onSave,
    required this.onDiscard,
  });

  @override
  State<ReviewShiftSheet> createState() => _ReviewShiftSheetState();
}

class _ReviewShiftSheetState extends State<ReviewShiftSheet> {
  late DateTime _startTime;
  late DateTime _endTime;
  late int _breakMinutes;
  late bool _isBreakPaid;
  late double _tips;
  late double _hourlyRate;
  late String _jobRole;
  late String _notes;
  late List<Expense> _expenses;

  final TextEditingController _tipsController = TextEditingController();
  final TextEditingController _breakController = TextEditingController();
  final TextEditingController _notesController = TextEditingController();

  final timeFormat = DateFormat('HH:mm');
  final dateFormat = DateFormat('dd.MM.yyyy');

  @override
  void initState() {
    super.initState();
    _startTime = widget.shift.startTime;
    _endTime = widget.shift.endTime ?? DateTime.now();
    _breakMinutes = widget.shift.breakDurationMinutes;
    _isBreakPaid = widget.shift.isBreakPaid;
    _tips = widget.shift.tips;
    _hourlyRate = widget.shift.hourlyRate;
    _jobRole = widget.shift.jobRole;
    _notes = widget.shift.notes;
    _expenses = List.from(widget.shift.expenses);

    _tipsController.text = _tips > 0 ? _tips.toString() : '';
    _breakController.text = _breakMinutes.toString();
    _notesController.text = _notes;
  }

  @override
  void dispose() {
    _tipsController.dispose();
    _breakController.dispose();
    _notesController.dispose();
    super.dispose();
  }

  Shift get _currentShift {
    return Shift(
      id: widget.shift.id,
      startTime: _startTime,
      endTime: _endTime,
      breakDurationMinutes: _breakMinutes,
      isBreakPaid: _isBreakPaid,
      hourlyRate: _hourlyRate,
      jobRole: _jobRole,
      tips: _tips,
      notes: _notes,
      expenses: _expenses,
    );
  }

  void _showAddExpenseDialog() {
    String title = '';
    double amount = 0.0;
    String category = 'נסיעות';

    showDialog(
      context: context,
      builder: (ctx) => AlertDialog(
        title: const Text('הוסף הוצאה למשמרת'),
        content: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            TextField(
              decoration: const InputDecoration(labelText: 'תיאור (למשל מונית, אוכל)'),
              onChanged: (val) => title = val,
            ),
            const SizedBox(height: 8),
            TextField(
              decoration: const InputDecoration(labelText: 'סכום (₪)'),
              keyboardType: const TextInputType.numberWithOptions(decimal: true),
              onChanged: (val) => amount = double.tryParse(val) ?? 0.0,
            ),
          ],
        ),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(ctx),
            child: const Text('ביטול'),
          ),
          FilledButton(
            onPressed: () {
              if (title.isNotEmpty && amount > 0) {
                setState(() {
                  _expenses.add(Expense(title: title, amount: amount, category: category));
                });
                Navigator.pop(ctx);
              }
            },
            child: const Text('הוסף'),
          ),
        ],
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    final computed = _currentShift;

    return Directionality(
      textDirection: TextDirection.rtl,
      child: Container(
        padding: EdgeInsets.only(
          top: 20,
          left: 20,
          right: 20,
          bottom: MediaQuery.of(context).viewInsets.bottom + 20,
        ),
        decoration: BoxDecoration(
          color: Theme.of(context).colorScheme.surface,
          borderRadius: const BorderRadius.vertical(top: Radius.circular(28)),
        ),
        child: SingleChildScrollView(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              // Header
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Text(
                    'סיכום ועריכת משמרת',
                    style: Theme.of(context).textTheme.titleLarge?.copyWith(fontWeight: FontWeight.bold),
                  ),
                  IconButton(
                    icon: const Icon(Icons.close),
                    onPressed: () => Navigator.pop(context),
                  ),
                ],
              ),

              const SizedBox(height: 12),

              // Summary Highlight Card
              Card(
                color: Theme.of(context).colorScheme.primaryContainer,
                shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
                child: Padding(
                  padding: const EdgeInsets.all(16.0),
                  child: Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          Text('שכר נטו סופי', style: Theme.of(context).textTheme.labelMedium),
                          Text(
                            '₪${computed.netPay.toStringAsFixed(1)}',
                            style: TextStyle(
                              fontSize: 26,
                              fontWeight: FontWeight.w900,
                              color: Theme.of(context).colorScheme.primary,
                            ),
                          ),
                        ],
                      ),
                      Column(
                        crossAxisAlignment: CrossAxisAlignment.end,
                        children: [
                          Text('שעות נטו', style: Theme.of(context).textTheme.labelMedium),
                          Text(
                            '${computed.netDurationHours.toStringAsFixed(2)} שעות',
                            style: const TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
                          ),
                        ],
                      ),
                    ],
                  ),
                ),
              ),

              const SizedBox(height: 16),

              // Time & Date Pickers
              Row(
                children: [
                  Expanded(
                    child: ListTile(
                      contentPadding: EdgeInsets.zero,
                      title: const Text('כניסה', style: TextStyle(fontSize: 12)),
                      subtitle: Text(
                        '${dateFormat.format(_startTime)} ${timeFormat.format(_startTime)}',
                        style: const TextStyle(fontWeight: FontWeight.bold),
                      ),
                    ),
                  ),
                  Expanded(
                    child: ListTile(
                      contentPadding: EdgeInsets.zero,
                      title: const Text('יציאה', style: TextStyle(fontSize: 12)),
                      subtitle: Text(
                        '${dateFormat.format(_endTime)} ${timeFormat.format(_endTime)}',
                        style: const TextStyle(fontWeight: FontWeight.bold),
                      ),
                    ),
                  ),
                ],
              ),

              const SizedBox(height: 12),

              // Break Duration & Break Type
              Row(
                children: [
                  Expanded(
                    child: TextField(
                      controller: _breakController,
                      decoration: const InputDecoration(
                        labelText: 'הפסקה (בדקות)',
                        border: OutlineInputBorder(),
                      ),
                      keyboardType: TextInputType.number,
                      onChanged: (val) {
                        setState(() {
                          _breakMinutes = int.tryParse(val) ?? 0;
                        });
                      },
                    ),
                  ),
                  const SizedBox(width: 12),
                  Expanded(
                    child: Container(
                      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                      decoration: BoxDecoration(
                        border: Border.all(color: Theme.of(context).colorScheme.outlineVariant),
                        borderRadius: BorderRadius.circular(8),
                      ),
                      child: Row(
                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
                        children: [
                          const Flexible(
                            child: Text(
                              'הפסקה בתשלום',
                              style: TextStyle(fontSize: 12),
                              overflow: TextOverflow.ellipsis,
                            ),
                          ),
                          Switch(
                            value: _isBreakPaid,
                            onChanged: (val) => setState(() => _isBreakPaid = val),
                          ),
                        ],
                      ),
                    ),
                  ),
                ],
              ),

              const SizedBox(height: 12),

              // Tips Input
              TextField(
                controller: _tipsController,
                decoration: const InputDecoration(
                  labelText: 'טיפים נוספים (₪)',
                  border: OutlineInputBorder(),
                  prefixIcon: Icon(Icons.monetization_on),
                ),
                keyboardType: const TextInputType.numberWithOptions(decimal: true),
                onChanged: (val) {
                  setState(() {
                    _tips = double.tryParse(val) ?? 0.0;
                  });
                },
              ),

              const SizedBox(height: 12),

              // Job Role Selector
              DropdownButtonFormField<String>(
                value: _jobRole,
                decoration: const InputDecoration(
                  labelText: 'תפקיד',
                  border: OutlineInputBorder(),
                ),
                items: widget.availableRoles.map((role) {
                  return DropdownMenuItem(value: role, child: Text(role));
                }).toList(),
                onChanged: (val) {
                  if (val != null) setState(() => _jobRole = val);
                },
              ),

              const SizedBox(height: 16),

              // Expenses Section
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Text(
                    'הוצאות שקוזזו (${_expenses.length})',
                    style: const TextStyle(fontWeight: FontWeight.bold),
                  ),
                  TextButton.icon(
                    onPressed: _showAddExpenseDialog,
                    icon: const Icon(Icons.add, size: 16),
                    label: const Text('הוסף הוצאה'),
                  ),
                ],
              ),

              if (_expenses.isNotEmpty)
                ..._expenses.map((exp) => ListTile(
                      contentPadding: EdgeInsets.zero,
                      leading: const Icon(Icons.receipt, size: 18),
                      title: Text(exp.title),
                      trailing: Row(
                        mainAxisSize: MainAxisSize.min,
                        children: [
                          Text('-₪${exp.amount.toStringAsFixed(1)}', style: const TextStyle(color: Colors.red)),
                          IconButton(
                            icon: const Icon(Icons.delete, size: 18),
                            onPressed: () {
                              setState(() {
                                _expenses.remove(exp);
                              });
                            },
                          ),
                        ],
                      ),
                    )),

              const SizedBox(height: 16),

              // Bottom Actions
              Row(
                children: [
                  Expanded(
                    child: OutlinedButton(
                      onPressed: () {
                        widget.onDiscard();
                        Navigator.pop(context);
                      },
                      style: OutlinedButton.styleFrom(
                        foregroundColor: Colors.red,
                        padding: const EdgeInsets.symmetric(vertical: 14),
                      ),
                      child: const Text('מחק'),
                    ),
                  ),
                  const SizedBox(width: 12),
                  Expanded(
                    flex: 2,
                    child: FilledButton(
                      onPressed: () {
                        widget.onSave(_currentShift);
                        Navigator.pop(context);
                      },
                      style: FilledButton.styleFrom(
                        padding: const EdgeInsets.symmetric(vertical: 14),
                      ),
                      child: const Text('שמור משמרת'),
                    ),
                  ),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}

File: app/src/main/java/com/example/shiftly/ui/components/ReviewShiftSheet.kt
package com.example.shiftly.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.shiftly.data.model.Expense
import com.example.shiftly.data.model.Shift
import com.example.ui.theme.ShiftlyCoral
import com.example.ui.theme.ShiftlyGreen
import java.text.SimpleDateFormat
import java.util.*

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ReviewShiftSheet(
    shift: Shift,
    availableRoles: List<String>,
    onDismiss: () -> Unit,
    onSave: (Shift) -> Unit,
    onDiscard: () -> Unit
) {
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)

    val timeFormat = remember { SimpleDateFormat("HH:mm", Locale.getDefault()) }
    val dateFormat = remember { SimpleDateFormat("dd.MM.yyyy", Locale.getDefault()) }

    var jobRole by remember { mutableStateOf(shift.jobRole) }
    var breakMinutesText by remember { mutableStateOf(shift.breakDurationMinutes.toString()) }
    var isBreakPaid by remember { mutableStateOf(shift.isBreakPaid) }
    var tipsText by remember { mutableStateOf(if (shift.tips > 0) shift.tips.toString() else "") }
    var hourlyRateText by remember { mutableStateOf(shift.hourlyRate.toString()) }
    var notes by remember { mutableStateOf(shift.notes) }

    var expenses by remember { mutableStateOf(shift.expenses) }
    var showAddExpenseDialog by remember { mutableStateOf(false) }

    // Start / End adjustments
    var startTimeMs by remember { mutableLongStateOf(shift.startTime) }
    var endTimeMs by remember { mutableLongStateOf(shift.endTime ?: System.currentTimeMillis()) }

    val breakMinutes = breakMinutesText.toIntOrNull() ?: 0
    val tips = tipsText.toDoubleOrNull() ?: 0.0
    val hourlyRate = hourlyRateText.toDoubleOrNull() ?: shift.hourlyRate

    val previewShift = remember(startTimeMs, endTimeMs, breakMinutes, isBreakPaid, tips, hourlyRate, expenses, jobRole, notes) {
        Shift(
            id = shift.id,
            startTime = startTimeMs,
            endTime = endTimeMs,
            breakDurationMinutes = breakMinutes,
            isBreakPaid = isBreakPaid,
            expenses = expenses,
            tips = tips,
            jobRole = jobRole,
            notes = notes,
            hourlyRate = hourlyRate
        )
    }

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        dragHandle = { BottomSheetDefaults.DragHandle() },
        containerColor = MaterialTheme.colorScheme.surface
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp)
                .padding(bottom = 32.dp)
        ) {
            // Title & Date Header
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column {
                    Text(
                        text = "סיכום ועריכת משמרת",
                        style = MaterialTheme.typography.titleLarge,
                        fontWeight = FontWeight.Bold
                    )
                    Text(
                        text = dateFormat.format(Date(startTimeMs)),
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }

                IconButton(
                    onClick = onDiscard,
                    modifier = Modifier.testTag("discard_shift_button")
                ) {
                    Icon(
                        Icons.Default.DeleteOutline,
                        contentDescription = "מחק משמרת",
                        tint = MaterialTheme.colorScheme.error
                    )
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            LazyColumn(
                modifier = Modifier
                    .fillMaxWidth()
                    .weight(1f, fill = false),
                verticalArrangement = Arrangement.spacedBy(14.dp)
            ) {
                // Time range card
                item {
                    Card(
                        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f)),
                        shape = RoundedCornerShape(16.dp),
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Column(modifier = Modifier.padding(14.dp)) {
                            Text(
                                text = "זמני משמרת",
                                style = MaterialTheme.typography.labelLarge,
                                fontWeight = FontWeight.SemiBold
                            )
                            Spacer(modifier = Modifier.height(8.dp))
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween,
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                // Start Time
                                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                                    Text(
                                        text = "כניסה",
                                        style = MaterialTheme.typography.labelMedium,
                                        color = MaterialTheme.colorScheme.onSurfaceVariant
                                    )
                                    Text(
                                        text = timeFormat.format(Date(startTimeMs)),
                                        style = MaterialTheme.typography.titleLarge,
                                        fontWeight = FontWeight.Bold
                                    )
                                    Row {
                                        IconButton(
                                            onClick = { startTimeMs -= 15 * 60 * 1000 },
                                            modifier = Modifier.size(32.dp)
                                        ) {
                                            Icon(Icons.Default.Remove, contentDescription = "מינוס 15 דק", modifier = Modifier.size(16.dp))
                                        }
                                        IconButton(
                                            onClick = { startTimeMs += 15 * 60 * 1000 },
                                            modifier = Modifier.size(32.dp)
                                        ) {
                                            Icon(Icons.Default.Add, contentDescription = "פלוס 15 דק", modifier = Modifier.size(16.dp))
                                        }
                                    }
                                }

                                Icon(
                                    Icons.Default.ArrowForward,
                                    contentDescription = null,
                                    tint = MaterialTheme.colorScheme.primary
                                )

                                // End Time
                                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                                    Text(
                                        text = "יציאה",
                                        style = MaterialTheme.typography.labelMedium,
                                        color = MaterialTheme.colorScheme.onSurfaceVariant
                                    )
                                    Text(
                                        text = timeFormat.format(Date(endTimeMs)),
                                        style = MaterialTheme.typography.titleLarge,
                                        fontWeight = FontWeight.Bold
                                    )
                                    Row {
                                        IconButton(
                                            onClick = { endTimeMs -= 15 * 60 * 1000 },
                                            modifier = Modifier.size(32.dp)
                                        ) {
                                            Icon(Icons.Default.Remove, contentDescription = "מינוס 15 דק", modifier = Modifier.size(16.dp))
                                        }
                                        IconButton(
                                            onClick = { endTimeMs += 15 * 60 * 1000 },
                                            modifier = Modifier.size(32.dp)
                                        ) {
                                            Icon(Icons.Default.Add, contentDescription = "פלוס 15 דק", modifier = Modifier.size(16.dp))
                                        }
                                    }
                                }
                            }
                        }
                    }
                }

                // Job role selector
                item {
                    Text(
                        text = "תפקיד / מקום עבודה",
                        style = MaterialTheme.typography.labelLarge,
                        fontWeight = FontWeight.SemiBold
                    )
                    Spacer(modifier = Modifier.height(6.dp))
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        availableRoles.take(4).forEach { role ->
                            val selected = role == jobRole
                            FilterChip(
                                selected = selected,
                                onClick = { jobRole = role },
                                label = { Text(role) },
                                leadingIcon = if (selected) {
                                    { Icon(Icons.Default.Check, contentDescription = null, modifier = Modifier.size(16.dp)) }
                                } else null
                            )
                        }
                    }
                }

                // Break section & Paid toggle
                item {
                    Card(
                        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f)),
                        shape = RoundedCornerShape(16.dp),
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Column(modifier = Modifier.padding(14.dp)) {
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween,
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                Column {
                                    Text(
                                        text = "הפסקה בתשלום",
                                        style = MaterialTheme.typography.bodyLarge,
                                        fontWeight = FontWeight.SemiBold
                                    )
                                    Text(
                                        text = if (isBreakPaid) "ההפסקה נחשבת בשכר" else "ההפסקה מקוזזת משעות העבודה",
                                        style = MaterialTheme.typography.bodySmall,
                                        color = MaterialTheme.colorScheme.onSurfaceVariant
                                    )
                                }
                                Switch(
                                    checked = isBreakPaid,
                                    onCheckedChange = { isBreakPaid = it },
                                    modifier = Modifier.testTag("break_paid_switch")
                                )
                            }

                            Spacer(modifier = Modifier.height(10.dp))

                            OutlinedTextField(
                                value = breakMinutesText,
                                onValueChange = { breakMinutesText = it },
                                label = { Text("משך הפסקה (בדקות)") },
                                leadingIcon = { Icon(Icons.Default.Coffee, contentDescription = null) },
                                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
                                singleLine = true,
                                modifier = Modifier.fillMaxWidth().testTag("break_minutes_input")
                            )
                        }
                    }
                }

                // Rate & Tips
                item {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(12.dp)
                    ) {
                        OutlinedTextField(
                            value = hourlyRateText,
                            onValueChange = { hourlyRateText = it },
                            label = { Text("תעריף שעתי (₪)") },
                            leadingIcon = { Icon(Icons.Default.AttachMoney, contentDescription = null) },
                            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
                            singleLine = true,
                            modifier = Modifier.weight(1f).testTag("hourly_rate_input")
                        )

                        OutlinedTextField(
                            value = tipsText,
                            onValueChange = { tipsText = it },
                            label = { Text("טיפים (₪)") },
                            leadingIcon = { Icon(Icons.Default.MonetizationOn, contentDescription = null) },
                            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
                            singleLine = true,
                            modifier = Modifier.weight(1f).testTag("tips_input")
                        )
                    }
                }

                // Expenses Section
                item {
                    Card(
                        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f)),
                        shape = RoundedCornerShape(16.dp),
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Column(modifier = Modifier.padding(14.dp)) {
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween,
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                Text(
                                    text = "הוצאות משמרת (נסיעות, אוכל וכו')",
                                    style = MaterialTheme.typography.labelLarge,
                                    fontWeight = FontWeight.SemiBold
                                )
                                TextButton(
                                    onClick = { showAddExpenseDialog = true },
                                    modifier = Modifier.testTag("add_expense_button")
                                ) {
                                    Icon(Icons.Default.Add, contentDescription = null, modifier = Modifier.size(18.dp))
                                    Spacer(modifier = Modifier.width(4.dp))
                                    Text("הוסף הוצאה")
                                }
                            }

                            if (expenses.isEmpty()) {
                                Text(
                                    text = "אין הוצאות רשומות למשמרת זו",
                                    style = MaterialTheme.typography.bodySmall,
                                    color = MaterialTheme.colorScheme.onSurfaceVariant
                                )
                            } else {
                                expenses.forEach { exp ->
                                    Row(
                                        modifier = Modifier
                                            .fillMaxWidth()
                                            .padding(vertical = 4.dp),
                                        horizontalArrangement = Arrangement.SpaceBetween,
                                        verticalAlignment = Alignment.CenterVertically
                                    ) {
                                        Column {
                                            Text(text = exp.title, fontWeight = FontWeight.Medium)
                                            Text(
                                                text = exp.category,
                                                style = MaterialTheme.typography.bodySmall,
                                                color = MaterialTheme.colorScheme.onSurfaceVariant
                                            )
                                        }
                                        Row(verticalAlignment = Alignment.CenterVertically) {
                                            Text(
                                                text = "-₪${String.format(Locale.US, "%.1f", exp.amount)}",
                                                color = ShiftlyCoral,
                                                fontWeight = FontWeight.Bold
                                            )
                                            IconButton(
                                                onClick = { expenses = expenses.filter { it.id != exp.id } },
                                                modifier = Modifier.size(32.dp)
                                            ) {
                                                Icon(Icons.Default.Close, contentDescription = "מחק", modifier = Modifier.size(16.dp))
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }

                // Notes
                item {
                    OutlinedTextField(
                        value = notes,
                        onValueChange = { notes = it },
                        label = { Text("הערות") },
                        leadingIcon = { Icon(Icons.Default.Notes, contentDescription = null) },
                        modifier = Modifier.fillMaxWidth().testTag("notes_input")
                    )
                }

                // Live Summary Calculation Box
                item {
                    Card(
                        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.primaryContainer),
                        shape = RoundedCornerShape(16.dp),
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Column(modifier = Modifier.padding(16.dp)) {
                            Text(
                                text = "סיכום שכר משוער",
                                style = MaterialTheme.typography.labelLarge,
                                fontWeight = FontWeight.Bold,
                                color = MaterialTheme.colorScheme.onPrimaryContainer
                            )
                            Spacer(modifier = Modifier.height(10.dp))

                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween
                            ) {
                                Text("שעות נטו לתשלום:")
                                Text(
                                    "${String.format(Locale.US, "%.2f", previewShift.netDurationHours)} שעות",
                                    fontWeight = FontWeight.Bold
                                )
                            }
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween
                            ) {
                                Text("שכר בסיס (ברוטו):")
                                Text(
                                    "₪${String.format(Locale.US, "%.1f", previewShift.grossEarnings)}",
                                    fontWeight = FontWeight.Bold
                                )
                            }
                            if (previewShift.tips > 0) {
                                Row(
                                    modifier = Modifier.fillMaxWidth(),
                                    horizontalArrangement = Arrangement.SpaceBetween
                                ) {
                                    Text("טיפים:")
                                    Text(
                                        "+₪${String.format(Locale.US, "%.1f", previewShift.tips)}",
                                        fontWeight = FontWeight.Bold,
                                        color = ShiftlyGreen
                                    )
                                }
                            }
                            if (previewShift.totalExpenses > 0) {
                                Row(
                                    modifier = Modifier.fillMaxWidth(),
                                    horizontalArrangement = Arrangement.SpaceBetween
                                ) {
                                    Text("הוצאות שקוזזו:")
                                    Text(
                                        "-₪${String.format(Locale.US, "%.1f", previewShift.totalExpenses)}",
                                        fontWeight = FontWeight.Bold,
                                        color = ShiftlyCoral
                                    )
                                }
                            }

                            Divider(modifier = Modifier.padding(vertical = 8.dp))

                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween,
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                Text(
                                    "סה״כ לתשלום נטו:",
                                    style = MaterialTheme.typography.titleMedium,
                                    fontWeight = FontWeight.Bold
                                )
                                Text(
                                    "₪${String.format(Locale.US, "%.1f", previewShift.netPay)}",
                                    style = MaterialTheme.typography.titleLarge,
                                    fontWeight = FontWeight.ExtraBold,
                                    color = MaterialTheme.colorScheme.primary
                                )
                            }
                        }
                    }
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            // Action Buttons: Save or Cancel
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                OutlinedButton(
                    onClick = onDismiss,
                    modifier = Modifier.weight(1f).testTag("cancel_review_button")
                ) {
                    Text("ביטול")
                }

                Button(
                    onClick = { onSave(previewShift) },
                    modifier = Modifier.weight(1.5f).testTag("save_shift_button")
                ) {
                    Icon(Icons.Default.Check, contentDescription = null)
                    Spacer(modifier = Modifier.width(6.dp))
                    Text("שמור משמרת")
                }
            }
        }
    }

    // Add Expense Dialog
    if (showAddExpenseDialog) {
        var expTitle by remember { mutableStateOf("") }
        var expAmountText by remember { mutableStateOf("") }
        var expCategory by remember { mutableStateOf("נסיעות") }
        val categories = listOf("נסיעות", "אוכל", "ציוד", "אחר")

        AlertDialog(
            onDismissRequest = { showAddExpenseDialog = false },
            title = { Text("הוספת הוצאה למשמרת") },
            text = {
                Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                    OutlinedTextField(
                        value = expTitle,
                        onValueChange = { expTitle = it },
                        label = { Text("תיאור (למשל: מונית חזור)") },
                        singleLine = true,
                        modifier = Modifier.fillMaxWidth()
                    )
                    OutlinedTextField(
                        value = expAmountText,
                        onValueChange = { expAmountText = it },
                        label = { Text("סכום ב-₪") },
                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
                        singleLine = true,
                        modifier = Modifier.fillMaxWidth()
                    )
                    Text("קטגוריה:", style = MaterialTheme.typography.labelMedium)
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(6.dp)
                    ) {
                        categories.forEach { cat ->
                            FilterChip(
                                selected = cat == expCategory,
                                onClick = { expCategory = cat },
                                label = { Text(cat, fontSize = 12.sp) }
                            )
                        }
                    }
                }
            },
            confirmButton = {
                Button(
                    onClick = {
                        val amount = expAmountText.toDoubleOrNull() ?: 0.0
                        if (expTitle.isNotBlank() && amount > 0) {
                            expenses = expenses + Expense(
                                title = expTitle,
                                amount = amount,
                                category = expCategory
                            )
                            showAddExpenseDialog = false
                        }
                    }
                ) {
                    Text("הוסף")
                }
            },
            dismissButton = {
                TextButton(onClick = { showAddExpenseDialog = false }) {
                    Text("ביטול")
                }
            }
        )
    }
}
